codegenerator/cli/templates/static/codegen/src/eventFetching/PartitionedFetchState.res

```diff
@@ -22,181 +17,92 @@ let make = (
+    ~isFetchingAtHead=false,
+  )
 
   {
     maxAddrInPartition,
-    partitions,
+    partitions: [partition],
     endBlock,
     startBlock,
     logger,
   }
 }
 
+exception InvalidFetchState({message: string})
+
 let registerDynamicContracts = (
   {partitions, maxAddrInPartition, endBlock, startBlock, logger}: t,
-  dynamicContractRegistration: FetchState.dynamicContractRegistration,
+  dynamicContracts: array<TablesStatic.DynamicContractRegistry.t>,
   ~isFetchingAtHead,
 ) => {
-  let newestPartitionIndex = partitions->Array.length - 1
-  let newestPartition = switch partitions[newestPartitionIndex] {
-  | Some(p) => p
-  | None => Js.Exn.raiseError("Unexpected: No partitions in PartitionedFetchState")
-  }
-
-  let partitions = if newestPartition->FetchState.getNumContracts < maxAddrInPartition {
-    let updated =
-      newestPartition->FetchState.registerDynamicContract(
-        dynamicContractRegistration,
-        ~isFetchingAtHead,
-      )
-    partitions->Utils.Array.setIndexImmutable(newestPartitionIndex, updated)
-  } else {
-    let newPartition = FetchState.make(
-      ~partitionId=partitions->Array.length,
-      ~startBlock=dynamicContractRegistration.registeringEventBlockNumber,
-      ~logger,
-      ~staticContracts=[],
-      ~dynamicContractRegistrations=dynamicContractRegistration.dynamicContracts,
-      ~isFetchingAtHead,
+  let fetchState = switch partitions {
+  | [fetchState] => fetchState
+  | _ =>
+    raise(
+      InvalidFetchState({
+        message: "Unexpected: Invalid fetchState in PartitionedFetchState",
+      }),
     )
-    partitions->Array.concat([newPartition])
   }
 
-  if Env.Benchmark.shouldSaveData {
-    Benchmark.addSummaryData(
-      ~group="Other",
-      ~label="Num partitions",
-      ~value=partitions->Array.length->Int.toFloat,
-    )
+  {
+    partitions: [
+      fetchState->FetchState.registerDynamicContract(dynamicContracts, ~isFetchingAtHead),
+    ],
+    maxAddrInPartition,
+    endBlock,
+    startBlock,
+    logger,
   }
-
-  {partitions, maxAddrInPartition, endBlock, startBlock, logger}
 }
 
-exception UnexpectedPartitionDoesNotExist(partitionId)
-
 /**
 Updates partition at given id with given values and checks to see if it can be merged into its next register.
 Returns Error if the partition/node with given id cannot be found (unexpected)
 */
-let update = (self: t, ~id: id, ~latestFetchedBlock, ~newItems, ~currentBlockHeight, ~chain) => {
-  switch self.partitions[id.partitionId] {
-  | Some(partition) =>
-    partition
-    ->FetchState.setFetchedItems(
-      ~id=id.fetchStateId,
-      ~latestFetchedBlock,
-      ~newItems,
-      ~currentBlockHeight,
-    )
-    ->Result.map(updatedPartition => {
+let setQueryResponse = (
+  self: t,
+  ~query: FetchState.query,
+  ~latestFetchedBlock,
+  ~newItems,
+  ~currentBlockHeight,
+  ~chain,
+) => {
+  let partitionId = switch query {
+  | PartitionQuery({partitionId})
+  | MergeQuery({partitionId}) => partitionId
+  }
```

Comment:
Maybe query should keep shared fields outside of the variant constr payload

Reply:
I'll consider this suggestion, when I work on partition kind.